SWLOR.Game.Server/Service/Skill.cs

```diff
@@ -66,7 +66,7 @@ public static void GiveSkillXP(uint player, SkillType skill, int xp)
             {
                 SendMessageToPC(player, $"You earned {details.Name} skill experience. ({xp})");
             }
-            else
+            else if (debtRemoved == 0)
```

Comment:
On line 61 there's a removal of the XP debt but with this change, the player object won't be saved to the DB. I think we need to do that right before the return here to ensure we don't miss that change. I'd recommend doing it here versus after 61 to reduce the number of saves we make during a normal XP gain.

Reply:
I think you might have this the wrong way around - /before/ this change it wasn't being saved to the DB if all our XP was used to pay down debt, because we were returning.  This change means that if we've cleared any XP debt we don't return early, and go through to the save line further down. 

If there's no XP debt and no XP, then we will skip the DB save.